src/main/java/org/jabref/gui/preferences/customentrytypes/CustomEntryTypesTab.java

```diff
@@ -151,20 +151,23 @@ private void setupFieldsTable() {
         fieldNameColumn.setEditable(true);
         fieldNameColumn.setOnEditCommit(
                 (TableColumn.CellEditEvent<FieldViewModel, String> event) -> {
+                    // This makes the displayed name consistent to org.jabref.model.entry.field.Field #getDisplayName()
                     String newFieldValue = StringUtil.capitalizeFirst(event.getNewValue());
                     UnknownField field = (UnknownField) event.getRowValue().getField();
                     EntryTypeViewModel selectedEntryType = viewModel.selectedEntryTypeProperty().get();
                     ObservableList<FieldViewModel> entryFields = selectedEntryType.fields();
+                    // The second predicate will check if the user input the original field name or doesn't edit anything after double click
                     boolean fieldExists = entryFields.stream().anyMatch(fieldViewModel ->
-                            fieldViewModel.nameProperty().getValue().equalsIgnoreCase(newFieldValue));
+                            fieldViewModel.nameProperty().getValue().equalsIgnoreCase(newFieldValue) && !newFieldValue.equals(field.getDisplayName()));
```

Comment:
The new condition does not make use of ´fieldViewModel `

I think, following code matches your intention:

```java
boolean fieldExists = !newFieldValue.equals(field.getDisplayName()) && entryFields.stream().anyMatch(fieldViewModel ->
                            fieldViewModel.nameProperty().getValue().equalsIgnoreCase(newFieldValue));
```

Reply:
You're right. I shouldn't put it in the `anyMatch()` method when streaming